test/StressTestMetrics/Program.cs

```diff
@@ -0,0 +1,104 @@
+        private static readonly Counter<long> Counter = MyMeter.CreateCounter<long>("counter");
+        private static string[] dimensionValues = new string[] { "DimVal1", "DimVal2", "DimVal3", "DimVal4", "DimVal5", "DimVal6", "DimVal7", "DimVal8", "DimVal9", "DimVal10" };
+        private static Random random = new Random();
+
+        public static void Main(string[] args)
+        {
+            long numberOfMetricWriters = Environment.ProcessorCount;
+            long maxWritesPerWriter = 10000000;
+            var writes = new long[numberOfMetricWriters];
+            using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                .AddSource("TestMeter")
+                .Build();
+            Stopwatch sw = new Stopwatch();
+            sw.Start();
+
+            Parallel.Invoke(
+                () =>
+                {
+                    var watch = new Stopwatch();
+                    while (true)
+                    {
+                        long totalWrites = 0;
+                        for (int i = 0; i < numberOfMetricWriters; i++)
+                        {
+                            totalWrites += writes[i];
+                        }
+
+                        watch.Restart();
+                        Thread.Sleep(1000);
+                        watch.Stop();
+
+                        long newTotalWrites = 0;
+                        for (int i = 0; i < numberOfMetricWriters; i++)
+                        {
+                            newTotalWrites += writes[i];
+                        }
+
+                        var writesPerSec = (newTotalWrites - totalWrites) / (watch.ElapsedMilliseconds / 1000.0);
+                        Console.Title = $"Writes (Million/Sec): {writesPerSec / 1000000}";
+
+                        if (totalWrites > numberOfMetricWriters * 10000000)
+                        {
+                            break;
+                        }
+                    }
+                }, () =>
+                {
+                    Parallel.For(0, Environment.ProcessorCount, i
+                    =>
+                    {
+                        Console.WriteLine($"Metric push by {i} started.");
+                        while (true)
+                        {
+                            // 10 * 10 * 10 = 1000 unique combination is produced
+                            // which is well within the current hard-coded cap of
+                            // 2000.
+                            var tag1 = new KeyValuePair<string, object>("DimName1", dimensionValues[random.Next(0, 10)]);
+                            var tag2 = new KeyValuePair<string, object>("DimName2", dimensionValues[random.Next(0, 10)]);
+                            var tag3 = new KeyValuePair<string, object>("DimName3", dimensionValues[random.Next(0, 10)]);
+                            Counter?.Add(100, tag1, tag2, tag3);
+                            writes[i]++;
+                            if (writes[i] > maxWritesPerWriter)
```

Comment:
This looks a bit weird, could this be moved to the while condition?

Reply:
e.g. I think the semantic here is:

```csharp
for (int j = 0; j++; j < maxWritesPerWriter)
{
    this.Counter.Add(100, tag1, tag2, tag3);
    writes[i] = j;
}
```

Reply:
Stepping back, do we need to end the loop at all (or we want it to keep running)?

Reply:
yes for now. It ends when each writer writes 10M times.

In future, we can modify this (or have a separate one) to be running forever until cancelled explicitly, and validate that the metrics calculated are correct.